server/services/ads.js

```diff
@@ -2,18 +2,44 @@ module.exports = function initialize(params) {
   const knex = params.knex;
   const util = params.util;
 
-  function listAds(loggedIn, limit, offset, domain, position, location) {
-    let query = knex('ads').where({}).orderBy('created_at', 'desc');
-    if (limit !== undefined) query = query.limit(limit);
-    if (offset !== undefined) query = query.offset(offset);
+  function listAds(loggedIn, limit, offset, domain, position, location, sorting) {
+    let query = knex('ads');
+    let answers = knex('answers');
+
+    switch(sorting) {
+      case 'created_at_asc':
+        query.orderBy('created_at', 'asc');
+        break;
+      case 'answers_desc':
+        // Ads with most answers first. If equal, newest first.
+        answers.select('ad_id').count('* as count').groupBy('ad_id').as('answers');
+        query.leftOuterJoin(answers, 'answers.ad_id', 'ads.id').orderByRaw('count DESC NULLS LAST').orderBy('created_at', 'desc');
+        break;
+      case 'answers_asc':
+        // Ads with least answers first. If equal, newest first.
+        answers.select('ad_id').count('* as count').groupBy('ad_id').as('answers');
+        query = query.leftOuterJoin(answers, 'answers.ad_id', 'ads.id').orderByRaw('count ASC NULLS FIRST').orderBy('created_at', 'desc');
+        break;
+      case 'newest_answer_desc':
+        // Ads with the newest answer first. If equal, newest first.
+        answers.select('ad_id').max('created_at as newest_answer').groupBy('ad_id').as('answers');
+        query.leftOuterJoin(answers, 'answers.ad_id', 'ads.id').orderByRaw('newest_answer DESC NULLS LAST').orderBy('created_at', 'desc');
+        break;
+      default:
```

Comment:
Maybe add here explicitly a fall through `case 'created_at_desc'` so it's more greppable